front-end/src/screens/Settings/username.tsx

```diff
@@ -70,30 +78,61 @@ const Username = ({ className }:Props): JSX.Element => {
 	};
 
 	return (
-		<Form standalone={false} className={className}>
+		<Form standalone={false} className={className} onSubmit={handleSubmit(handleSubmitForm)}>
 			<Form.Group>
 				<Form.Field width={10}>
 					<label>Username</label>
+					{editing ?
+						<input
+							className={errors.username ? 'error' : ''}
+							defaultValue={username || ''}
+							name='username'
+							placeholder='username'
+							ref={register({ maxLength:30, minLength:3, pattern: /^[A-Za-z0-9._-]*$/, required: true })}
+							type='text'
+						/> :
+						<div className='current-username'>{username}</div>
+					}
+					{(errors.username as FieldError)?.type === 'maxLength' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_MAXLENGTH_ERROR}</span>}
+					{(errors.username as FieldError)?.type === 'minLength' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_MINLENGTH_ERROR}</span>}
+					{(errors.username as FieldError)?.type === 'pattern' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_PATTERN_ERROR}</span>}
+					{(errors.username as FieldError)?.type === 'required' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_REQUIRED_ERROR}</span>}
+				</Form.Field>
+				{!editing && <Form.Field width={6}>
+					<label>&nbsp;</label>
```

Comment:
Haven't seen those before. It's used as a spacer as far as I can see. We should user a spacer div (here and in all other places) then rather than an html `label` that is used for usability.

Reply:
added spacer div